builtin/game/misc.lua

```diff
@@ -125,6 +125,19 @@ function core.get_player_radius_area(player_name, radius)
 end
 
 
+function core.is_valid_pos(pos)
+	local mapgen_limit = tonumber(core.settings:get("mapgen_limit"))
```

Comment:
This value is fixed per-world so should be fetched once and cached, for performance.